### PR TITLE
Add examples for Gradle projects and the CommandAPI Kotlin-DSL as a Gradle project

### PR DESCRIPTION
**Pull request by Sytm**
Noteworthy is that I changed some functions in the KotlinDSL to inline even though they do not benefit from that because that way, the Kotlin DSL does not need to be shaded into the final plugin because everything is inlined.

### CODE REVIEW

File: docssrc/src/kotlinintro.md [hunk 1]
```diff
@@ -249,43 +167,19 @@ dependencies {
...
 
 <div class="multi-pre">
 
 ```groovy,build.gradle
-shadowJar {
-    dependencies {
-        include dependency("org.jetbrains.kotlin:kotlin-stdlib")
-    }
+kotlin {
+	jvmToolchain(16)
```
Comment (DerEchtePilz) [inline docssrc/src/kotlinintro.md:176]: A tab is used here. Please only use spaces in the documentation.

File: docssrc/src/kotlinintro.md [hunk 2]
```diff
@@ -249,43 +167,19 @@ dependencies {
...
 }
 ```
 
 ```kotlin,build.gradle.kts
-shadowJar {
-    dependencies {
-        include dependency("org.jetbrains.kotlin:kotlin-stdlib")
-    }
+kotlin {
+	jvmToolchain(16)
```
Comment (DerEchtePilz) [inline docssrc/src/kotlinintro.md:182]: A tab is used here. Please only use spaces in the documentation.

**Review by DerEchtePilz — CHANGES_REQUESTED** [review]
In general this looks good.
I've mentioned two points that need to be fixed - otherwise the tab will get displayed as 8 spaces in the documentation and that looks awful.

Also, if you're changing the Kotlin version in the documentation, please also change it in the root `pom.xml`. There is a `properties` section with `kotlin.version`. This value is used in all the Kotlin modules.

**Comment by Sytm** [comment]
On the topic of tabs and spaces I edited the .editorconfig and added a spaces rule for the docssrc folder.

**Review by DerEchtePilz — APPROVED** [review]

**Comment by Sytm** [comment]
Also the task "Build CommandAPI example Bukkit plugins in examples/ folder" looks broken (and needs changes because of the additional gradle examples)

**Comment by DerEchtePilz** [comment]
I approved those changes for now - I am still not sure about the Spigot thingy. Using Paper doesn't do anything bad so I guess we'll ask @JorelAli to approve that.

Good job on the PR though!

**Comment by Sytm** [comment]
I am working on this shell script atm too to also add the gradle examples

**Comment by Sytm** [comment]
I could change the used toolchain in the examples to match the one used for the build to fix this failure

File: examples/bukkit/kotlindsl/src/main/kotlin/io/github/jorelali/Main.kt [hunk 3]
```diff
@@ -1,15 +1,10 @@
...
 
-import dev.jorel.commandapi.CommandAPI
-import dev.jorel.commandapi.CommandAPIBukkitConfig
 import org.bukkit.plugin.java.JavaPlugin
 
 class Main : JavaPlugin() {
 
 	override fun onEnable() {
-		CommandAPI.onLoad(CommandAPIBukkitConfig(this))
-		CommandAPI.onEnable()
```
Comment (DerEchtePilz) [inline examples/bukkit/kotlindsl/src/main/kotlin/io/github/jorelali/Main.kt:11]: Why did you remove these two lines? I feel like they shoud stay because the examples are supposed to present a possible way of creating a plugin using the CommandAPI.

Comment (Sytm) [reply examples/bukkit/kotlindsl/src/main/kotlin/io/github/jorelali/Main.kt:11]: I removed them because they are only required when actually shading the plugin, but we are not. (https://commandapi.jorel.dev/8.8.0/shading.html).

Comment (DerEchtePilz) [reply examples/bukkit/kotlindsl/src/main/kotlin/io/github/jorelali/Main.kt:11]: Thanks for pointing that out, I forgot what `commandapi-bukkit-core` is doing...
